CONCEPTS.md

```diff
@@ -0,0 +1,62 @@
+# Concepts
+
+## Table of Contents
+- [Metric](#metric)
+- [Tag Key and Tag Value](#tag-key-and-tag-value)
+- [Timestamp](#timestamp)
+- [Timeseries](#timeseries)
+- [Gauge](#gauge)
+- [Counter](#counter)
+- [Timer](#timer)
+- [Metric Registry](#metric-registry)
+- [Reporter](#reporter)
+- [Time Window](#time-window)
+- [Monoid](#monoid)
+
+### Metric
+A quantifiable characteristic of an application that is being measured. Metrics have numeric values that are either 64-bit signed integers or 64-bit double-precision floating-point values. Metrics have names that are strings and should meaningfully describe the quantity being measured. Some examples -
+* `webserver.rps` the number of requests per second being processed by a web server
+* `webserver.request.latency` time taken (latency, cumulative or percentiles) to process requests by a web server
+
+### Tag Key and Tag Value
+A tag (aka dimension) is a pair that consists of a key and a value. Tags describe the context for which a metric is being recorded. For example, `host=webserver1.domain.com`. Here, `host` is a tag key and `webserver1.domain.com` is the tag value. Other examples could include `url`, `resource-path`, `response-code`, `application-id`, `region`, or `availability-zone`.
```

Comment:
Do most timeseries dbs allow dash in tag names? If so, we are good, otherwise remove dash from example.

Reply:
yes, most of the ones i'm aware of allow this.